packages/flutter/lib/src/widgets/app.dart

```diff
@@ -769,6 +770,53 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
 
   // BUILDER
 
+  bool _debugCheckLocalizations(Locale appLocale) {
+    assert(() {
+      final Set<Type> unsupportedTypes =
+        _localizationsDelegates.map<Type>((LocalizationsDelegate<dynamic> delegate) => delegate.type).toSet();
+      for (LocalizationsDelegate<dynamic> delegate in _localizationsDelegates) {
+        if (!unsupportedTypes.contains(delegate.type))
+          continue;
+        if (delegate.isSupported(appLocale))
+          unsupportedTypes.remove(delegate.type);
+      }
+      if (unsupportedTypes.isEmpty)
+        return true;
+
+      // Currently the Cupertino library only provides english localizations.
+      // Remove this when https://github.com/flutter/flutter/issues/23847
+      // is fixed.
+      if (listEquals(unsupportedTypes.map((Type type) => type.toString()).toList(), <String>['CupertinoLocalizations']))
+        return true;
+
+      final StringBuffer message = StringBuffer();
+      message.writeln('\u2550' * 8);
+      message.writeln(
+        'Warning: This application\'s locale, $appLocale, is not supported by all of its\n'
+        'localization delegates.'
+      );
+      for (Type unsupportedType in unsupportedTypes) {
+        // Currently the Cupertino library only provides english localizations.
+        // Remove this when https://github.com/flutter/flutter/issues/23847
+        // is fixed.
+        if (unsupportedType.toString() == 'CupertinoLocalizations')
+          continue;
+        message.writeln(
+          '> A delegate that supports $unsupportedType was not found.'
```

Comment:
You're right, the message could be clearer. The two-part structure of the message is needed because other delegate types (CupertinoLocalizations, WidgetLocalizations, app specific localizations) might be included in the "not supported" list. But the message could definitely be clearer.

I'll change it to:
```
Warning: This application's locale, foo_BAR, is not supported by all of its
localization delegates.
> A MaterialLocalizations delegate that supports the foo_BAR locale was not found.
See https://flutter.io/tutorials/internationalization/ for more
information about configuring an app's locale, supportedLocales,
and localizationsDelegates parameters.
```

Reply:
Nice. That's much clearer. 👍